spec/outputs/lumberjack_spec.rb

```diff
@@ -1 +1,89 @@
+# encoding: utf-8
+require "logstash/outputs/Lumberjack"
+require "logstash/event"
 require "logstash/devutils/rspec/spec_helper"
+require "lumberjack/server"
+require "flores/pki"
+require "stud/temporary"
+require "fileutils"
+
+describe "Sending events" do
+  let(:batch_size) { Flores::Random.integer(20..100) }
+  let(:batch_payload) do
+    batch = []
+    batch_size.times do |n|
+      batch << LogStash::Event.new({ "message" => "foobar #{n}" })
+    end
+    batch
+  end
+
+  let(:number_of_crash) { Flores::Random.integer(1..3) }
+  let(:certificate) { Flores::PKI.generate }
+  let(:certificate_file_crt) { Stud::Temporary.pathname }
+  let(:certificate_file_key) { Stud::Temporary.pathname }
+  let(:port) { Flores::Random.integer(1024..65535) }
+  let(:host) { "127.0.0.1" }
+  let(:queue) { [] }
+
+  let(:client_options) {
+    {
+      "hosts" => [host],
+      "port" => port,
+      "ssl_certificate" => certificate_file_crt,
+      "flush_size" => 10#batch_size # flush at the end of the payload
```

Comment:
I usually test with static variable before adding random values this is left over.